### PR TITLE
cnf-tests: Bump dependencies

### PR DESCRIPTION
**Pull request by liornoy**
This PR bumps K8S to 1.27 and also bumping the pinned vendors (ptp, sriov, metallb, nto)
Some code modifications were required to fix breaking changes.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift-kni/cnf-features-deploy
 
-go 1.19
+go 1.20
```
Comment (ffromani) [inline go.mod:3]: do we want to bump the golang version (do we need to bump the toolchain as well?) in the same commit and in the same PR?

**Comment by Tal-or** [comment]
May we close https://github.com/openshift-kni/cnf-features-deploy/pull/1575?

File: go.mod (hunk 1 above)
Comment (SchSeba) [reply go.mod:3]: good question maybe we can have the go bump in a separate PR

Comment (Tal-or) [reply go.mod:3]: @liornoy please go update from this PR
BTW it seems like it needs more work anyway (just a headsup for #1634)
```
Trying to pull registry.ci.openshift.org/openshift/release:golang-1.20...
Warning: Pull failed, retrying in 5s ...
Trying to pull registry.ci.openshift.org/openshift/release:golang-1.20...
Warning: Pull failed, retrying in 5s ...
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: initializing source docker://registry.ci.openshift.org/openshift/release:golang-1.20: reading manifest golang-1.20 in registry.ci.openshift.org/openshift/release: manifest unknown
```

**Comment by liornoy** [comment]
> Shall we close #1575?

yes. I the deletion of RTE deps and NTO bump is included here.

**Comment by zeeke** [comment]
/approve
LGTM. Waiting for someone from compute team to review

**Review by ffromani — COMMENTED** [review]
/lgtm
/hold

compute-related changes seem seem straightforward

**Comment by ffromani** [comment]
holding to let other reviewers chime in, feel free to remove anytime

**Pull request merged**